lib/BannerPlugin.js

```diff
@@ -26,21 +69,20 @@ class BannerPlugin {
 	}
 
 	apply(compiler) {
-		const options = this.options;
-		const banner = this.banner;
+		const options = this.options,
+			banner = this.banner;
 
 		compiler.plugin("compilation", (compilation) => {
 			compilation.plugin("optimize-chunk-assets", (chunks, callback) => {
 				chunks.forEach((chunk) => {
 					if(options.entryOnly && !chunk.isInitial()) return;
-
 					chunk.files
 						.filter(ModuleFilenameHelpers.matchObject.bind(undefined, options))
-						.forEach((file) =>
-							compilation.assets[file] = new ConcatSource(
-								banner, "\n", compilation.assets[file]
-							)
-						);
+						.forEach((file) => {
+							const chunkHash = chunk.renderedHash || chunk.hash;
+							const comment = interpolate(banner, file, compilation.hash, chunkHash);
```

Comment:
please use the interpolation function which is also used by other stuff: `compilation.getPath`